ras_backstage/resources/reporting_units/get_reporting_unit.py

```diff
@@ -17,6 +21,40 @@ class GetReportingUnit(Resource):
     @staticmethod
     def get(ru_ref):
         logger.info('Retrieving reporting unit details', ru_ref=ru_ref)
+
         reporting_unit = party_controller.get_party_by_ru_ref(ru_ref)
-        logger.info('Successfully retrieved party details', ru_ref=ru_ref)
-        return make_response(jsonify(reporting_unit), 200)
+
+        surveys = get_surveys_for_reporting_unit(reporting_unit)
+
+        cases = case_controller.get_cases_by_business_party_id(reporting_unit['id'])
+        case_collection_exercise_ids = [case['caseGroup']['collectionExerciseId'] for case in cases]
+
+        for survey in surveys:
+            ces = collection_exercise_controller.get_collection_exercises_by_survey(survey['id'])
+            survey['collection_exercises'] = [ce for ce in ces if ce['id'] in case_collection_exercise_ids]
```

Comment:
Can we do the check in one loop instead of looping through again?
```
survey['collection_exercises']  = [ce for ce in ces if ce['id'] in case_collection_exercise_ids and parse_date(ce['scheduledStartDateTime']) < now]
```